regress/pesterTests/Uninstall.Tests.ps1

```diff
@@ -0,0 +1,235 @@
+
+        $RegReadKeyPerm = ([System.UInt32] [System.Security.AccessControl.RegistryRights]::ReadKey.value__)
+        $RegFullControlPerm = [System.UInt32] [System.Security.AccessControl.RegistryRights]::FullControl.value__        
+
+        #only validate owner and ACEs of the registry
+        function ValidateRegistryACL {
+            param([string]$RegPath, $Ownersid = $adminsSid, $IdAcls)
+            Test-Path -Path $RegPath | Should Be $true                      
+            $myACL = Get-ACL $RegPath
+            $OwnerSid = Get-UserSid -User $myACL.Owner
+            $OwnerSid.Equals($Ownersid) | Should Be $true
+            $myACL.Access | Should Not Be $null
+            $CAPABILITY_SID = "S-1-15-3-1024-1065365936-1281604716-3511738428-1654721687-432734479-3232135806-4053264122-3456934681"            
+            $nonPropagate = $myACL.Access | ? {($_.PropagationFlags -eq ([System.Security.AccessControl.PropagationFlags]::None)) -and ($_.IdentityReference -ine $CAPABILITY_SID)}
+
+            foreach ($a in $nonPropagate) {
+                $findItem = $IdAcls | ? {
+                    ($a.IdentityReference -eq (Get-UserAccount -UserSid ($_.Identity))) -and `
+                    ($a.IsInherited -eq $_.IsInherited) -and `
+                    ($a.AccessControlType -eq ([System.Security.AccessControl.AccessControlType]::Allow)) -and  `
+                    ($a.PropagationFlags -eq ([System.Security.AccessControl.PropagationFlags]::None) -and  `
+                    (([System.Int32]$a.RegistryRights.value__) -eq ($_.RegistryRights))) 
+                }
+                $findItem | Should Not Be $null
+            }         
+        }    
+    }    
+
+    Context "$tC - Validate Openssh binary files" {
+        BeforeAll {
+            if(-not $Windowsbox)
+            {
+                $binaries =  $null
+                $dataFile = $null
+                $dataFile1 = $null
+                return
+            }
+            $tI=1
+            $binaries =  @(
+                @{
+                    Name = 'sshd.exe'
+                },
+                @{
+                    Name = 'ssh.exe'
+                },
+                @{
+                    Name = 'ssh-agent.exe'
+                },
+                @{
+                    Name = 'ssh-add.exe'
+                },
+                @{
+                    Name = 'sftp.exe'
+                },
+                @{
+                    Name = 'sftp-server.exe'
+                },
+                @{
+                    Name = 'scp.exe'
+                },
+                @{
+                    Name = 'ssh-shellhost.exe'
+                },
+                @{
+                    Name = 'ssh-agent.exe'
+                },
+                @{
+                    Name = 'ssh-keyscan.exe'
+                }
+            )
+            $dataFile =  @(
```

Comment:
looks like this is not being used?

Reply:
they are needed for window in box. The tests will validate all these file does not exist on the machine.

Reply:
yes, you are right.